types/single-spa-react/index.d.ts

```diff
@@ -1,29 +1,32 @@
 // Type definitions for single-spa-react 2.8
 // Project: https://github.com/CanopyTax/single-spa-react
 // Definitions by: Garrett Smith <https://github.com/Garrett-Smith-iq>
+//                 Chris Dopuch <https://github.com/chrisdopuch>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import * as React from 'react';
-import * as ReactDOM from 'react-dom';
+import * as React from "react";
+import * as ReactDOM from "react-dom";
 
 export = SingleSpaReact;
-declare function SingleSpaReact(opts: SingleSpaReact.Options): SingleSpaReact.Lifecycles;
+declare function SingleSpaReact(
+    opts: SingleSpaReact.Options
+): SingleSpaReact.Lifecycles;
 
 declare namespace SingleSpaReact {
-	interface Options {
-		React: typeof React;
-		ReactDOM: typeof ReactDOM;
-		rootComponent?: React.ComponentClass;
-		loadRootComponent?: () => Promise<React.ComponentClass>;
-		domElementGetter?: () => Element;
-		suppressComponentDidCatchWarning?: boolean;
-		parcelCanUpdate?: boolean;
-	}
+    interface Options {
+        React: typeof React;
+        ReactDOM: typeof ReactDOM;
+        rootComponent?: React.ComponentClass<any, any>;
```

Comment:
Specifying props and state as `any` here avoids the error you get when using `strictFunctionTypes` in tsconfig.json and passing a component with custom props as `rootComponent`
(I tried with a generic at first, but dtslint warned me that it was a useless generic only used once -- so instead I just loosened the types of props and state to be more permissive)